pandas/tests/frame/methods/test_replace.py

```diff
@@ -1496,6 +1496,14 @@ def test_replace_list_with_mixed_type(
         result = obj.replace(box(to_replace), value)
         tm.assert_equal(result, expected)
 
+    @pytest.mark.parametrize("val", [2, np.nan, 2.0])
+    def test_replace_value_none_dtype_numeric(self, val):
+        # GH#48231
+        df = DataFrame({"a": [1, val]})
+        result = df.replace(val, None)
+        expected = DataFrame({"a": [1, np.nan]})
```

Comment:
So just confirming this result shouldn't be this?

```
In [1]: pd.Series([1.0, np.nan]).replace(np.nan, value=None)
Out[1]:
0     1.0
1    None
dtype: object

In [2]: pd.__version__
Out[2]: '1.4.4'
```

Reply:
Not sure if this is really what we want...

Wanted to check if ci passes like this and go from there

Reply:
Gotcha. IMO if we go with the current result here, I think we should document that the original `df/ser` needs to be `object` type for `None` to appear.

Anecdotally, I've definitely seen `replace(np.nan, None)` or `replace(np.nan, "")` and people being surprised that `np.nan` remains.

Reply:
cc @mroeschke This would restore the 1.4.x behavior for now

Reply:
IIRC we had special logic for the case where the user specifically wants to replace one NA-like value with another (in this case np.nan->None).  The non-special logic would do something like:

```
def replace(self, old, new):
    if is_valid_na_for_dtype(new, self.dtype):
        new = self. _standardize_fill_value(new)
```

which for float64 dtype would take the None back to np.nan, consistent with what we do with _every other_ setitem-like method.  But in this case it is what the user specifically doesn't want.

Not sure if this is helpful, but there it is.

Reply:
Yeah this would make more sense imo too, but would break previous behavior and fail a couple of tests too